### ISSUE TITLE
Failed to initialize warp storage.

### ISSUE THREAD

**Issue by Ironic8b498** [TheE/MyWarp#94]
I get this error at bootup, running Minecraft 1.9.2, with these plugins:
* Autorank 3.7.2
* BungeeTabListPlus_BukkitBridge-2.4.3
* CoreProtect_2.12.0
* CoreProtect-Anti-Xray-1.3
* Essentials-2.x-SNAPSHOT-13
* EssentialsChat-2.x-SNAPSHOT-13
* mywarp-bukkit-3.0-SNAPSHOT-all-111
* MyZ
* NoPlugins-2.0.1
* PandoraChatFilter-11.7
* PermissionsEx-1.23.4
* ProtocolLib-1.9- 23
* Vault-1.5.6
* worldedit-bukkit-6.1.1-SNAPSHOT-dist-3463
* worldguard-legacy-6.1.2-SNAPSHOT-dist- 1653

MyWarp runs on the rest of my servers, just not sure why not on this one, MyZ and  ProtocolLib is only ran on this server.

````
[22:13:02] [Server thread/INFO]: [MyWarp] Enabling MyWarp v3.0-SNAPSHOT+jenkins-b111.git-07f5f2e
[22:13:02] [Server thread/INFO]: Flyway 4.0 by Boxfuse
[22:13:04] [Server thread/ERROR]: [MyWarp] Failed to initialize warp storage.
me.taylorkelly.mywarp.warp.storage.StorageInitializationException: Failed to execute migration process.
	at me.taylorkelly.mywarp.warp.storage.WarpStorageFactory.createInitialized(WarpStorageFactory.java:108) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.MyWarp.initialize(MyWarp.java:107) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.MyWarpPlugin.onEnable(MyWarpPlugin.java:103) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:291) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:340) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.loadPlugin(CraftServer.java:361) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.enablePlugins(CraftServer.java:321) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at net.minecraft.server.v1_9_R1.MinecraftServer.t(MinecraftServer.java:411) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at net.minecraft.server.v1_9_R1.MinecraftServer.l(MinecraftServer.java:376) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at net.minecraft.server.v1_9_R1.MinecraftServer.a(MinecraftServer.java:331) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at net.minecraft.server.v1_9_R1.DedicatedServer.init(DedicatedServer.java:269) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at net.minecraft.server.v1_9_R1.MinecraftServer.run(MinecraftServer.java:527) [spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_72-internal]
Caused by: me.taylorkelly.mywarp.internal.flyway.core.api.FlywayException: Unable to obtain Jdbc connection from DataSource
	at me.taylorkelly.mywarp.internal.flyway.core.internal.util.jdbc.JdbcUtils.openConnection(JdbcUtils.java:56) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.internal.flyway.core.Flyway.execute(Flyway.java:1314) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.internal.flyway.core.Flyway.repair(Flyway.java:1098) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.warp.storage.WarpStorageFactory.createInitialized(WarpStorageFactory.java:105) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	... 13 more
Caused by: java.sql.SQLException: invalid database address: jdbc:h2:/var/games/minecraft/servers/AaronsWorld/plugins/MyWarp/warps
	at org.sqlite.JDBC.createConnection(JDBC.java:74) ~[spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at org.sqlite.JDBC.connect(JDBC.java:64) ~[spigot-1.9.jar:git-Spigot-bc01c3a-0cd6711]
	at java.sql.DriverManager.getConnection(DriverManager.java:664) ~[?:1.8.0_72-internal]
	at java.sql.DriverManager.getConnection(DriverManager.java:208) ~[?:1.8.0_72-internal]
	at me.taylorkelly.mywarp.bukkit.util.jdbc.SingleConnectionDataSource.initiate(SingleConnectionDataSource.java:169) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.util.jdbc.SingleConnectionDataSource.getConnection(SingleConnectionDataSource.java:84) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.internal.flyway.core.internal.util.jdbc.JdbcUtils.openConnection(JdbcUtils.java:50) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.internal.flyway.core.Flyway.execute(Flyway.java:1314) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.internal.flyway.core.Flyway.repair(Flyway.java:1098) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.warp.storage.WarpStorageFactory.createInitialized(WarpStorageFactory.java:105) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	... 13 more
[22:13:04] [Server thread/ERROR]: [MyWarp] MyWarp is unable to continue and will be disabled.
[22:13:04] [Server thread/INFO]: [MyWarp] Disabling MyWarp v3.0-SNAPSHOT+jenkins-b111.git-07f5f2e
````

**Comment by TheE** [comment]
For some reason MyWarp fails to connect with the H2 database. Did MyWarp run on this server before?

Does MyWarp's plugin-folder exist and are there any files within it?

**Comment by Ironic8b498** [comment]
Yes it ran when it was 1.8.8 only updated the other day when MyZ was finally updated to 1.9.2. I did try last month but server crashed many times. I even deleted the mywarp folder to start all over and it can't write the files.

`[07:53:46] [Server thread/INFO]: [MyWarp] Enabling MyWarp v3.0-SNAPSHOT+jenkins-b111.git-07f5f2e
[07:53:46] [Server thread/ERROR]: [MyWarp] Failed to create the default configuration file ('/var/games/minecraft/servers/Walkin/plugins/MyWarp/config.yml'), using build-in defaults for all values.
java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively(Native Method) ~[?:1.8.0_72-internal]
	at java.io.File.createNewFile(File.java:1012) ~[?:1.8.0_72-internal]
	at me.taylorkelly.mywarp.bukkit.settings.BukkitSettings.reload(BukkitSettings.java:72) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.settings.BukkitSettings.<init>(BukkitSettings.java:63) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.BukkitPlatform.<init>(BukkitPlatform.java:68) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.MyWarpPlugin.onEnable(MyWarpPlugin.java:99) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:292) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:340) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.loadPlugin(CraftServer.java:361) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.enablePlugins(CraftServer.java:321) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.t(MinecraftServer.java:411) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.l(MinecraftServer.java:376) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.a(MinecraftServer.java:331) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.DedicatedServer.init(DedicatedServer.java:269) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.run(MinecraftServer.java:527) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_72-internal]
[07:53:46] [Server thread/WARN]: [MyWarp] Failed to access SQLite profile cache. Player names will be resolved from memory.
java.io.IOException: Failed to connect to cache file
	at me.taylorkelly.mywarp.internal.squirrelid.cache.SQLiteCache.<init>(SQLiteCache.java:67) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.SquirrelIdProfileCache.<init>(SquirrelIdProfileCache.java:64) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.BukkitPlatform.<init>(BukkitPlatform.java:70) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at me.taylorkelly.mywarp.bukkit.MyWarpPlugin.onEnable(MyWarpPlugin.java:99) [mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:292) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:340) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.loadPlugin(CraftServer.java:361) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.enablePlugins(CraftServer.java:321) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.t(MinecraftServer.java:411) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.l(MinecraftServer.java:376) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.a(MinecraftServer.java:331) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.DedicatedServer.init(DedicatedServer.java:269) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.run(MinecraftServer.java:527) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_72-internal]
Caused by: java.sql.SQLException: path to '/var/games/minecraft/servers/Walkin/plugins/MyWarp/profiles.db': '/var/games/minecraft/servers/Walkin/plugins/MyWarp' does not exist
	at org.sqlite.Conn.open(Conn.java:103) ~[spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.sqlite.Conn.<init>(Conn.java:57) ~[spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.sqlite.JDBC.createConnection(JDBC.java:77) ~[spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.sqlite.JDBC.connect(JDBC.java:64) ~[spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at java.sql.DriverManager.getConnection(DriverManager.java:664) ~[?:1.8.0_72-internal]
	at java.sql.DriverManager.getConnection(DriverManager.java:270) ~[?:1.8.0_72-internal]
	at me.taylorkelly.mywarp.internal.squirrelid.cache.SQLiteCache.<init>(SQLiteCache.java:63) ~[mywarp-bukkit-3.0-SNAPSHOT-all-%23111.jar:3.0-SNAPSHOT+jenkins-b111.git-07f5f2e]
	... 14 more
[07:53:46] [Server thread/ERROR]: Error occurred while enabling MyWarp v3.0-SNAPSHOT+jenkins-b111.git-07f5f2e (Is it up to date?)
java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:213) ~[spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at me.taylorkelly.mywarp.warp.storage.ConnectionConfiguration.<init>(ConnectionConfiguration.java:52) ~[?:?]
	at me.taylorkelly.mywarp.bukkit.settings.BukkitSettings.getRelationalStorageConfiguration(BukkitSettings.java:283) ~[?:?]
	at me.taylorkelly.mywarp.MyWarp.initialize(MyWarp.java:103) ~[?:?]
	at me.taylorkelly.mywarp.bukkit.MyWarpPlugin.onEnable(MyWarpPlugin.java:103) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:292) ~[spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:340) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.loadPlugin(CraftServer.java:361) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at org.bukkit.craftbukkit.v1_9_R1.CraftServer.enablePlugins(CraftServer.java:321) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.t(MinecraftServer.java:411) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.l(MinecraftServer.java:376) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.a(MinecraftServer.java:331) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.DedicatedServer.init(DedicatedServer.java:269) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at net.minecraft.server.v1_9_R1.MinecraftServer.run(MinecraftServer.java:527) [spigot-1.9.2.jar:git-Spigot-f6feaaa-f326992]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_72-internal]`

**Comment by TheE** [comment]
MyWarp cannot write to its configuration folder; I cannot fix this as this is an (strange) error within your setup.

Could you please create a folder named `MyWarp` within Bukkit's plugin-folder manually and ensure that the user that runs the minecraft server can read and write to it?

**Comment by Ironic8b498** [comment]
it created the files if I left folder there and deleted the files in side. so must be something to do with my OS, I'm running MineOS, been using for 3 years now and fist time I had this issue with any plugin.

**Comment by TheE** [comment]
Should be fixed now (duplicate of #104).
